controller/up.go

```diff
@@ -68,6 +135,9 @@ func compress(src string, buf io.Writer) error {
 			return err
 		}
 
+		// close the file to avoid hitting fd limit
+		_ = f.Close()
```

Comment:
if there's an error we should probably alert the user but complete anyways

Something like this (don't know what the API is for file off the top of my head)

```
		err = f.Close()
		if err != nil {
		    fmt.Printf("Failed to add %s", f.name) 
		}
```

Reply:
agreed